src/NServiceBus.AcceptanceTesting/Support/TaskExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿namespace NServiceBus.AcceptanceTesting.Support
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    static class TaskExtensions
+    {
+        public static Task Timebox(this IEnumerable<Task> tasks, TimeSpan timeoutAfter, string messageWhenTimeboxReached)
+        {
+            var taskCompletionSource = new TaskCompletionSource<object>();
+            var tokenSource = new CancellationTokenSource(timeoutAfter);
+            var registration = tokenSource.Token.Register(s =>
+            {
+                var tcs = (TaskCompletionSource<object>)s;
+                tcs.TrySetException(new Exception(messageWhenTimeboxReached));
```

Comment:
I wanted to keep it inline with the current behavior. Happy to change it if you think it is worth it